From 080ee441563b2ecb1ee5012b0885f2c9af51beb2 Mon Sep 17 00:00:00 2001
From: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date: Thu, 1 Oct 2020 00:29:31 +0900
Subject: [PATCH] Support python 3.9 and drop support python < 3.2

python 3.9 removes getiterator() of classes ElementTree , which are
obsoleted from python 3.2:
https://docs.python.org/3.9/whatsnew/3.9.html#removed

Change getiterator() to iter() introduced in python 3.2, and because of
this, drop support python < 3.2.
---
 catfish_lib/Builder.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/catfish_lib/Builder.py b/catfish_lib/Builder.py
index 9c8d478..5e74dcb 100644
--- a/catfish_lib/Builder.py
+++ b/catfish_lib/Builder.py
@@ -84,7 +84,7 @@ class Builder(Gtk.Builder):
         tree = ElementTree()
         tree.parse(filename)
 
-        ele_widgets = tree.getiterator("object")
+        ele_widgets = tree.iter("object")
         for ele_widget in ele_widgets:
             try:
                 name = ele_widget.attrib['id']
@@ -109,7 +109,7 @@ class Builder(Gtk.Builder):
             if connections:
                 self.connections.extend(connections)
 
-        ele_signals = tree.getiterator("signal")
+        ele_signals = tree.iter("signal")
         for ele_signal in ele_signals:
             self.glade_handler_dict.update(
                 {ele_signal.attrib["handler"]: None})
-- 
2.28.0

